src/components/Pagination/Paginator.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import PaginationRow from './PaginationRow';
+import { bindMethods, noop } from '../../common/helpers';
+import { PAGINATION_VIEW_TYPES } from './constants';
+
+class Paginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    bindMethods(this, ['handleFormSubmit']);
+
+    this.initPagination(props);
+
+    this.state = {
+      pagination: props.pagination,
+      pageChangeValue: props.pagination.page
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { pagination } = nextProps;
+    if (
+      this.props.pagination.page !== pagination.page ||
+      this.props.pagination.perPage !== pagination.perPage ||
+      this.props.pagination.perPageOptions.toString() !==
+        pagination.perPageOptions.toString()
+    ) {
+      this.setState({
+        pagination: pagination,
+        pageChangeValue: pagination.page
```

Comment:
Not a big thing but you should imho cast the values to Number here. Otherwise the state may alter between `1` and `"1"`.

Reply:
done